### PR TITLE
Add prerequisites table and update logic to handle it

### PR DESCRIPTION
**Pull request by IsharaEkanayaka**

### CODE REVIEW

File: database/migrations/2024_09_03_102824_create_course_modules_table.php [hunk 1]
```diff
@@ -15,7 +15,7 @@ public function up()
     {
         Schema::create('course_modules', function (Blueprint $table) {
             $table->id();  // Primary key, auto-incrementing
-            $table->foreignId('course_id')->constrained('courses');  // Foreign key to courses.id
+            $table->foreignId('course_id')->constrained('courses')->onDelete('cascade');  // Foreign key to courses.id
```
Comment (NuwanJ) [inline database/migrations/2024_09_03_102824_create_course_modules_table.php:18]: Doing this change needs to be in a separate migration file.

Comment (NuwanJ) [reply database/migrations/2024_09_03_102824_create_course_modules_table.php:18]: The reason is when we do release 2.1.0, this migration file is already ran

File: database/migrations/2024_10_08_195256_modify_course_modules_course_id_foreign.php
Comment (NuwanJ) [inline database/migrations/2024_10_08_195256_modify_course_modules_course_id_foreign.php:1]: @IsharaEkanayaka Correct, you got the concept !!!

**Review by NuwanJ — APPROVED** [review]

**Comment by NuwanJ** [comment]
@IsharaEkanayaka unit tests failing due to the `SQLite` database not supporting the `dropForeign` option in the test environment (Work with MySQL) Will you be able to see what we can to to bypass it while unit testing?

**Comment by IsharaEkanayaka** [comment]
> @IsharaEkanayaka unit tests failing due to the `SQLite` database not supporting the `dropForeign` option in the test environment (Work with MySQL) Will you be able to see what we can to to bypass it while unit testing?

can't we change testing database to MySQL

**Comment by NuwanJ** [comment]
I think it might not fit with the CI/CD pipeline to set up MySQL. The best option is to skip the migration file's code if DB is SQLite

**Comment by IsharaEkanayaka** [comment]
> I think it might not fit with the CI/CD pipeline to set up MySQL. The best option is to skip the migration file's code if DB is SQLite

got it. i will do it

**Review by NuwanJ — APPROVED** [review]
Approving, CI/CD issue will be resolved seperate

**Pull request merged**